test/libsolidity/semanticTests/inlineAssembly/transient_storage_selfdestruct.sol

```diff
@@ -0,0 +1,44 @@
+contract C {
+    function set(uint value) external {
+        assembly {
+            tstore(0, value)
+        }
+    }
+
+    function get() external view returns (uint value) {
+        assembly {
+            value := tload(0)
+        }
+    }
+
+    function terminate(address payable a) external {
+        selfdestruct(a);
+    }
+}
+
+contract D {
+    C public c;
+
+    constructor() {
+        c = new C();
+    }
+
+    function destroy() external {
+        c.set(42);
+        c.terminate(payable(address(this)));
+        assert(c.get() == 42);
+    }
+
+    function createAndDestroy() external {
+        c = new C();
+        c.set(42);
+        c.terminate(payable(address(this)));
+        assert(c.get() == 42);
+    }
+}
+// ====
+// EVMVersion: >=cancun
+// ----
+// constructor() ->
+// destroy() ->
+// createAndDestroy() ->
```

Comment:
This test only tests destruction in a separate transaction, which is less relevant. Your original test was also checking for transient storage changes when creation and destruction happens in the same transaction.

The point is to show that even in the corner case where the contract gets deleted, its transient storage is not cleared until the end of the transaction.

Reply:
But isn't `createAndDestroy` doing that?

Reply:
Ah, wait. Not sure how I missed that :) I guess thew new test just has less tangential stuff than the old one.

OK then, it's fine.